Introduce extensibility hooks across CRM, HRM, IMAP, Log, and UI components.

aminurislamarnob opened the pull request:
#### Changes proposed in this Pull Request:

**Feature:** Introduce extensibility hooks across CRM, HRM, IMAP, Log, and UI components.

This update adds new extensibility hooks throughout several core modules of the WP ERP system, specifically within the CRM, HRM, IMAP, Logging, and UI components. These hooks allow developers to more easily customize and extend the functionality of WP ERP via actions and filters, improving integration and modularity.

### Key Highlights:
- Added new hooks (actions and filters) in CRM, HRM, IMAP, Log, and UI components.
- Improved developer experience for plugin and extension authors.
- Enhanced flexibility for future customizations and integration scenarios.

iftakharul-islam reviewed (changes requested):
@aminurislamarnob, please check the reviews I mentioned for you. They just need a small prefix addition and a validation check.

aminurislamarnob commented on modules/crm/includes/functions-customer.php:
```diff
@@ -2697,6 +2702,21 @@ function erp_crm_save_email_activity( $email, $inbound_email_address ) {
+         */
+        if( apply_filters( 'erp_crm_skip_owner_email_notification', false, $to_email, $email ) ) {
+            // Update email counter
+            update_option( 'wp_erp_inbound_email_count', get_option( 'wp_erp_inbound_email_count', 0 ) + 1 );
```
@iftakharul-islam vai, I just reused ERP previously defined update option. Please check the last two lines of this current function `erp_crm_save_email_activity`

aminurislamarnob commented on modules/hrm/includes/Models/HrUser.php:
```diff
@@ -17,6 +17,12 @@ public function notes() {
     }
 
     public function getTable() {
+        global $wpdb;
```
@iftakharul-islam vai, Better I have removed core changes here for multisite check and introducing a filter hook for third-party plugin extensibility.

aminurislamarnob commented on modules/hrm/includes/functions-employee.php:
```diff
@@ -76,16 +76,17 @@ function erp_hr_get_employees( $args = [] ) {
 
     $cache_key_counts = 'erp-get-employees-count-' . md5( serialize( $args ) ) . " : $last_changed";
     $results_counts   = wp_cache_get( $cache_key_counts, 'erp' );
+    $usermeta_table = apply_filters( 'erp_usermeta_table_name', $wpdb->prefix . 'usermeta' );
```
@iftakharul-islam vai, I have added prefix for the hook.

aminurislamarnob commented:
> @aminurislamarnob, please check the reviews I mentioned for you. They just need a small prefix addition and a validation check.

@iftakharul-islam vai, Thanks for the heads-up! I’ve addressed the prefix updates and added the necessary validation check as suggested. Let me know if anything else needs tweaking.

Rubaiyat-E-Mohammad commented:
Getting this build issue @aminurislamarnob vai
<img width="720" height="562" alt="image" src="https://github.com/user-attachments/assets/858d6b77-dbab-4328-ac75-2d68b4cc1bb2" />

aminurislamarnob commented:
> Getting this build issue @aminurislamarnob vai <img alt="image" width="720" height="562" src="https://private-user-images.githubusercontent.com/79570623/485621843-858d6b77-dbab-4328-ac75-2d68b4cc1bb2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5ODc5MDAsIm5iZiI6MTc1Njk4NzYwMCwicGF0aCI6Ii83OTU3MDYyMy80ODU2MjE4NDMtODU4ZDZiNzctZGJhYi00MzI4LWFjNzUtMmQ2OGI0Y2MxYmIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTA0VDEyMDY0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxMjRlMTE2YTUzNjVkN2Q3YTgwMWI1YjUxZGUzNzIxYmJiMDBlNDJiMzhhMThmYWExZmNhZDkzMjhlNDkzOWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sV9i1tj4a23uzORuQMua4_hYBfsDjdOInxHzwpOUzLw">

@Rubaiyat-E-Mohammad vai, are you getting this error on the develop branch?

Rubaiyat-E-Mohammad commented:
Can you suggest some user scenarios where this PR can have effect? @aminurislamarnob vai

aminurislamarnob commented:
> Can you suggest some user scenarios where this PR can have effect? @aminurislamarnob vai
@Rubaiyat-E-Mohammad vai, there won’t be any changes to the default ERP features. We’ve only introduced hooks to improve extensibility. You just need to verify that the full default ERP process is still working correctly after our changes.
